tests/e2e.rs

```diff
@@ -1,130 +1,73 @@
 
-    let alice1 = alice0.receive(message0_bob, &alice_wallet).await.unwrap();
-    let bob1 = bob0.receive(message0_alice, &bob_wallet).await.unwrap();
+    let CreateActors {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = make_create_actors(&bitcoind, alice_balance, bob_balance, time_lock).await;
 
-    let message1_alice = alice1.next_message();
-    let message1_bob = bob1.next_message();
+    let Created { alice, bob } = run_create_protocol(&alice_wallet, alice, &bob_wallet, bob).await;
 
-    let alice2 = alice1.receive(message1_bob).unwrap();
-    let bob2 = bob1.receive(message1_alice).unwrap();
+    assert_eq!(alice.signed_TX_f, bob.signed_TX_f);
 
-    let message2_alice = alice2.next_message();
-    let message2_bob = bob2.next_message();
-
-    let alice3 = alice2.receive(message2_bob).unwrap();
-    let bob3 = bob2.receive(message2_alice).unwrap();
-
-    let message3_alice = alice3.next_message();
-    let message3_bob = bob3.next_message();
-
-    let alice4 = alice3.receive(message3_bob).unwrap();
-    let bob4 = bob3.receive(message3_alice).unwrap();
+    alice_wallet
+        .0
+        .send_raw_transaction(alice.signed_TX_f.clone())
+        .await
+        .unwrap();
+}
 
-    let message4_alice = alice4.next_message();
-    let message4_bob = bob4.next_message();
+#[tokio::test]
+async fn e2e_channel_update() {
+    let tc_client = testcontainers::clients::Cli::default();
+    let bitcoind = Bitcoind::new(&tc_client, "0.19.1").unwrap();
 
-    let alice5 = alice4.receive(message4_bob).unwrap();
-    let bob5 = bob4.receive(message4_alice).unwrap();
+    let time_lock = 1;
+    let (alice_balance, bob_balance) = (Amount::ONE_BTC, Amount::ONE_BTC);
 
-    let message5_alice = alice5.next_message(&alice_wallet).await.unwrap();
-    let message5_bob = bob5.next_message(&bob_wallet).await.unwrap();
+    let CreateActors {
+        alice,
+        alice_wallet,
+        bob,
+        bob_wallet,
+    } = make_create_actors(&bitcoind, alice_balance, bob_balance, time_lock).await;
 
-    let alice6 = alice5.receive(message5_bob, &alice_wallet).await.unwrap();
-    let bob6 = bob5.receive(message5_alice, &bob_wallet).await.unwrap();
+    let Created { alice, bob } = run_create_protocol(&alice_wallet, alice, &bob_wallet, bob).await;
 
-    assert_eq!(alice6.signed_TX_f, bob6.signed_TX_f);
+    assert_eq!(alice.signed_TX_f, bob.signed_TX_f);
 
     alice_wallet
         .0
-        .send_raw_transaction(alice6.signed_TX_f.clone())
+        .send_raw_transaction(alice.signed_TX_f.clone())
         .await
         .unwrap();
+
+    let UpdateActors { alice, bob } = make_update_actors(alice, bob);
+
+    let channel_update = ChannelUpdate::Pay(Amount::from_btc(0.5).unwrap());
+    let time_lock = 1;
+
+    let Updated { alice, bob } = run_update_protocol(alice, bob, channel_update, time_lock);
+
+    assert_eq!(alice.balance().unwrap().ours, bob.balance().unwrap().theirs);
```

Comment:
Shouldn't we test that the balance are as expected by the test and not just sync'd between the parties?

Reply:
That's a good point! I'll do that.